test/fixtures/schema.sql

```diff
@@ -305,7 +305,7 @@ SET search_path = postgrest, pg_catalog;
 
 CREATE TABLE auth (
     id character varying NOT NULL,
-    rolname name NOT NULL,
+    rolname name NOT NULL DEFAULT 'postgrest_test_author',
```

Comment:
Found the reason. I didn't completely understand how the tests worked and assumed the database (or at least the auth table) was truncated after every test. `post "/postgrest/users" [json| { "id": "jdoe"...` being in every test confused me so I wasn't catching any errors from that statement.